paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -29,34 +29,35 @@ using mkldnn::stream;
 using platform::to_void_cast;
 using platform::GetMKLDNNFormat;
 
-inline void GetWeightsTz(std::vector<int>& weights_tz, int groups,  // NOLINT
-                         bool is_conv3d) {
+constexpr int same_scale_mask = 0;
+constexpr int o_slice_mask = 1 << 0;                         // 1
+constexpr int g_slice_mask = 1 << 1;                         // 2
+constexpr int g_o_slice_mask = g_slice_mask | o_slice_mask;  // 3
+
+static int ComputeWeightsMask(int is_multi_channel, int g) {
+  if (is_multi_channel) {
+    return g > 1 ? g_o_slice_mask : o_slice_mask;
+  } else {
+    return same_scale_mask;
+  }
+}
+
+static int ComputeBiasMask(int is_multi_channel) {
+  if (is_multi_channel) {
+    return o_slice_mask;
+  } else {
+    return same_scale_mask;
+  }
```

Comment:
ComputeWeightsMask and ComputeBiasMask look very similar. Could you please extract most of their functionality into a ComputeMask(bool is_multi_channel, int multi_channel_mask) function? It'll look much simpler that way.
```C++
static int ComptueMask(bool is_multi_channel, int multi_channel_mask) {
  if (is_multi_channel) {
    return multi_channel_mask;
  } else {
    return same_scale_mask;
  }
}

static int ComputeWeightsMask(bool is_multi_channel, int g) {
  int multi_channel_mask = g > 1 ? g_o_slice_mask : o_slice_mask;
  return ComptueMask(is_multi_channel, multi_channel_mask);
}

static int ComputeBiasMask(int is_multi_channel) {
  return ComputeMask(is_multi_channel, o_slice_mask);
}

Reply:
Hi @Sand3r- I changed as you said. I think you are right, this way looks much simpler ~